js/src/hooks/useCroppedImageSelector/useCroppedImageSelector.js

```diff
@@ -0,0 +1,289 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useCallback, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import './useCroppedImageSelector.scss';
+
+/**
+ * @typedef {import('.~/hooks/types.js').ImageMedia} ImageMedia
+ */
+
+/**
+ * @typedef {Object} CroppedImageSelector
+ * @property {(preselectedId: number) => void} openSelector Function to open the selector modal.
+ */
+
+/**
+ * Calculates the maximum cropping size according to the fixed aspect ratio.
+ *
+ * @param {number} width The image width.
+ * @param {number} height The image height.
+ * @param {number} widthScale The scale of width.
+ * @param {number} heightScale The scale of height.
+ * @return {[number, number]} The tuple of cropped width and height.
+ */
+export function cropByFixedRatio( width, height, widthScale, heightScale ) {
+	const ratio = widthScale / heightScale;
+
+	if ( width / height > ratio ) {
+		width = Math.round( height * ratio );
+	} else {
+		height = Math.round( width / ratio );
+	}
+
+	return [ width, height ];
+}
```

Comment:
❓  I guess this just calculate the maximum cropping size but is not actually cropping the image.

💅  If so, I would rename the function as  `calculateMaxCropByFixedRatio` or something similar. IMO without reading the function internals `cropByFixedRatio` could be understood as we are actually cropping the image.

Reply:
Renamed in https://github.com/woocommerce/google-listings-and-ads/pull/1819/commits/d11fc0ec9a053e95a77aadb8dcfd6fca1f76bea3.